packages/astro/src/compiler/codegen/index.ts

```diff
@@ -47,6 +47,22 @@ interface CodeGenOptions {
   fileID: string;
 }
 
+interface HydrationAttributes {
+  method?: 'load' | 'idle' | 'visible';
+}
+
+/** Searches through attributes to extract hydration-rlated attributes */
+function findHydrationAttributes(attrs: Record<string, string>): HydrationAttributes {
+  let method: HydrationAttributes['method'];
+
+  for (const [key, val] of Object.entries(attrs)) {
+    if (!key.startsWith(':')) continue;
+    if ([':load', ':idle', ':visible'].includes(key) && val === 'true') method= key.slice(1) as HydrationAttributes['method'];
```

Comment:
I would move this:

```js
[':load', ':idle', ':visible'].includes(key)
```

Into a `Set` outside of this function. Free O(1) lookups :)

```js
const hydrationDirectives = new Set([':load', ':idle', ':visible']);

...

if(hydrationDirectives.has(key)) ...
```

Reply:
Good call, I shamelessly copy/pasted this from an early discussion around the hydration syntax :smile: